
\documentclass{article}
\usepackage{authblk}
\usepackage{tabularx}
\usepackage{float}

\newcolumntype{b}{X}
\newcolumntype{s}{>{\hsize=.2\hsize}X}

\title{Cryptic and not-so-cryptic species in the complex ``\textit{Holothuria
    (Thymiosycia) imaptiens}'' (Forrsk\r{a}ll, 1775) (Echinodermata:
  Holothuroidea: Holothuriidae)} \author[1]{Fran\c{c}ois
  Michonneau\thanks{Corresponding author: francois.michonneau@gmail.com}}
\author[1]{Gustav Paulay} \affil[1]{Florida Museum of Natural History,
  University of Florida, Gainesville, FL 32611-7800, USA}

\renewcommand\Authands{ and }


\begin{document}

<<setup, echo=FALSE, cache=FALSE, include=FALSE>>=
opts_chunk$set(fig.path='figures/latex-', cache.path='cache/latex-', dev="tikz",
               external=TRUE, fig.width=7, fig.height=4, fig.align="center")
read_chunk('code/impatiens_analysis.R')
read_chunk('code/impatiens_map.R')
@ 

<<init-phylo, echo=FALSE, cache=TRUE, include=FALSE>>=
@ 

<<init-map, echo=FALSE, cache=TRUE, include=FALSE>>=
@ 

\maketitle

\begin{abstract}
Blabla
\end{abstract}

\section{Introduction}

The failure to distinguish reproductively isolated species can stem from either
the choice of an inappropriate set of characters to distinguish species
otherwise morphologically variable, or from the absence of variation in
morphology but differentiation in other traits (e.g., ecology, chemical cues)
allows them to remain reproductively isolated \cite{Knowlton1993}.

For morphological characters to be appropriate to distinguish between
populations of reproductively isolated individuals, they require to evolve at a
rate of the same par as populations become species. They also must exhibit less
intraspecific than interspecific variation. Traits that are under sexual
selection usually meet these criteria. When species use visual clues to identify
their mates, taxonomists can use the same characters to discrimminate between
these populations of reproductively isolated individuals. However, in situations
where mate recognition does not involve visual clues (e.g., chemical
communication, sperm/egg recognition proteins), whether a particular
morphological trait evolves at the appropriate pace to distinguish between
species is governed by other evolutionary forces. These traits are, at best, a
by-product of reproductive isolation that do not guarantee to be effective in
discrimminating species.

Most marine invertebrate taxonomy results from the study of dry or preserved
specimens. With these methods, many delicate morphological characters and very
often color patterns are lost. Therefore, in groups characterized by hard parts,
taxonomists have relied on these skeletal elements to find characters that allow
them to differentiate species. For instance, in sea cucumbers, in particular in
the Holothuriidae, species-level taxonomy has entirely been focused on the shape
of ossicles, microscopic calcareous secretions found in many tissues. At least
until the late 1980's most species descriptions would ignore intraindividual
variation by figuring only ossicles found in the body wall of the species, and
generally ignored any intraspecific variation by illustrating the ossicle
assemblage of a single individual.

% who were the first authors to illustrate ossicles?

Forrsk\r{a}ll described \textit{Holothuria impatiens} in 1775 from the
collections he was making in Suez, Egypt. The original description is vague, and
could fit many other holothuriid species of the area, he did however include a
drawing of the species. Since then, the range of \textit{H. impatiens} has been
extended, and the species is now thought to be occurring in the Red Sea, the
entire Indo-West Pacific region, the Caribbean and the Mediterrean seas.

\textit{Holothuria impatiens} is the type species of the subgenus
\textit{Thymiosycia} one of the 18 subgenera that are forming the genus
\textit{Holothuria}. The subgenus \textit{Thymiosycia} was first erected by
Pearson \cite{Pearson1915} to include all holothuriid being characterized by the
absence of ``true pedicels'' and anal teeth, and the presence of ``papillae
scattered all over the body, generally on eminences''; and designed
\textit{H. impatiens} as the type species for the subgenus. In the most recent
revision of the family, Rowe \cite{Rowe1969} redefined \textit{Thymiosycia} to
only include species which are characterized by having well developed, regular
and symmetrical tables and buttons. The ossicle assemblage of
\textit{H. impatiens} therefore exemplifies the holothuriid ossicles.

\textit{Holothuria impatiens} is a common, locally fairly abundant, species
which can be found under rocks in shallow waters (in particular in lagoons and
back-reef habitats) but has been recorded down to 158~m (\cite{Samyn2013}).
Because of its ubiquituous distribution, it is one of the more studied species
of holothuriid with studies on its reproduction \cite{Harriot1985}, its
Cuvierian tubules \cite{Flammang2002}, \cite{Becker+Flammang2010}; its toxicity
\cite{Bakus1974}; its feeding preferences \cite{Roberts1982}; the chemical
composition \cite{Hampton1958}, statistical analysis \cite{Hampton1959},
ontogenic changes in its ossicles \cite{Cutress1996}; and its parasites
\cite{Martens1994}.  It has also been included in molecular phylogenies (e.g.,
\cite{Lacey2005}, \cite{Honey-Escandon2012}). \textit{Holothuria impatiens} is a
low-value commercial species but is reported to be fished in the Eastern Pacific
\cite{Toral-Granda2008}, Madagascar \cite{Conand2007}, and Palau
\cite{Pakoa2009}.

Despite its relative biological and commercial importance, the variation
observed in their color patterns and reported by previous workers (e.g.,
\cite[p.178]{Clark1921}, \cite{Rowe+Richmond2004}) has yet to be
investigated. In the present study, using a combination of morphological,
genetic and ecological data, we show that \textit{Holothuria impatiens}
represents a complex of at least 12 putative species. If some of these species
can be distinguished relatively easily from their live appearance, others are
harder to tease apart. This, in combination with the fact that several of these
species co-occur, and sometimes even share the same habitat, has contributed to
the failure to identify the species limits in this complex.

The development of genetic data and its application to species identification
have allowed the discovery of many species complexes. The DNA barcoding approach
of using an easy-to-amplify single locus to delineate species provides a
promising venue to speed up biodiversity documentation. Ideally the analysis of
the genetic data should delineate the biological species accurately with little
or no subjectivity, so that the process can be automated and used on large
datasets. However the biological processes generating the genetic
differentiation that can be used to delineate species, depends on a variety of
other factors than reproductive isolation that a single locus cannot
capture. Nonetheless, threshold based methods (e.g. \cite{}) or other methods
developped \cite{Pons2006}, \cite{Monaghan2009} to detect species limits are
relatively succesful. While the possibility of detecting accurately species
limits from a single marker remains contentious, using multiple lines of
evidence to assess species limits in species complexes detected from
single-locus dataset is widespread. The development of integrative taxonomy
which uses multiple lines of evidence to delinate species has recently seen more
formal development in how to determine the species limits based on available
evidence. The most recent developments favor finding a biological explanation
behind contradictory lines of evidence \cite{Schlick-Steiner2010}. For genetic
data, recent developments in Bayesian statistics allow for integrating genetic
signal from a variety of markers by invoking the multi-coalescent process and
also compare competing models.

The genetic differentiation observed in species previously thought to be found
across large geographic areas (e.g, circum-tropical species) has changed our
understanding of how biodiversity is distributed in the sea, with isolated
oceanic bassins and archipelagoes exhibiting much higher endemism rates than
previously thought (e.g., %find references for Caribbean, Red Sea, Medit
\cite{Meyer2005a}). If unraveling diversity among geographically isolated
populations affects estimates of global diversity, unraveling diversity among
sympatric populations affects not only local estimates of diversity, but also
challenges our understanding of the processes generating biodiversity.

Understanding the dynamics of diversification in the sea is challenged by the
fact that most marine invertebrates have a highly dispersive larval stage and
that oceans in general and the Indo-Pacific in particular do not seem to have
clear geographical barriers to dispersal. Yet, while some species have broad
geographical ranges encompassing most of the IWP province, others are more
limited, and some closely related species have allopatric distributions. The
frequent overlap in species distribution across the IWP challenges our current
understanding of the role of geographical isolation in speciation. It is
generally accepted, at least on land, that allopatry is the most common mode of
speciation. However, the combination of broadly overlapping ranges and high
diversity leaves seemingly few opportunities for allopatry in the
sea. Furthermore, recent geological and molecular evidence suggests that most of
the modern reef fauna was shaped during the Miocene. To account for this
apparent paradox, several alternative models have been proposed. One category
downplays the importance of the absence of gene flow (allopatry) and emphasizes
the role of selection (along environemental gradients, sexual selection, host
specificity) as the main driver behind reproductive isolation (see
\cite{Bowen2013}). Another category of models proposes that speciation occurred
in allopatry, but since then, important changes in species ranges and
distributions have obscurred the barriers responsible for the initiation of the
reproductive isolation.

Inferring the mode of speciation from current distributions in the Indo-Pacific
is most likely misleading as modern ranges in the Pacific are necessarily
relatively recent as sea levels have fluctuated greatly (ca. 100-150~m) and
repeatedly (ca. 100~k.y. periodicity) during the late Tertiary and early
Quarternary. These fluctuations have led to important habitat modifications and
changes in oceanic current patterns with consequences in connectivity,
population sizes and species ranges. These changes have made species
distributions in the Pacific highly dynamic (\cite{Paulay1990}) and are, in most
cases, too recent compared to the time frame needed for speciation. 
% However, speciation can be very rapid, e.g., Asterinid from NSW Byrne2012
For instance, the latest minimum sea level (120 to 135~m below present) was
18,000 years ago. During this period, circulation between Pacific and Indian
Oceans was significantly restricted, the Red Sea was most likely disconnected
from the Indian Ocean, and the South China sea was land-locked
(\cite{Veron1992}). Therefore, current distribution patterns reflect only
partially the biogeographical context under which the diversification of modern
reef-associated fauna has occurred.

The consequences of these fluctuations in sea level on the populations might
however still be visible in the present genetic diversity of the modern
fauna. For instance, the low sea levels caused an important reduction of the
connectivity between the Indian and Pacific oceans providing repeated
opportunity for population divergence between the two ocean bassins. While some
species retained a genetic signature of this episode of low connectivity in
their mitochondrial DNA (e.g., xxxxxx), many other lineages (including the most
widespread form of \textit{Holothuria impatiens}) show little geographic genetic
structuring across their range. This latter pattern can be explained by (1)
maintained connectivity during these low sea stands, as both oceans were still
connected (\cite{Voris2000}); (2) one of the two lineages was lost because of
local extinction or selective sweep; (3) the species recently expanded their
ranges and colonized one of the bassin very recently.


% Allopatry without clear geographic barriers (Diadema setosum, Lessios 2001)

% how to include peripatric speciation?
% say something about how these 2 models are not mutually exclusive

\section{Methods}

\subsection{Approach}

To estimate the extant of the diversity in the \textit{Holothuria impatiens}
complex, we first assessed differentiation observed in COI to make sense of the
poorly understood morphological variation that is seen in the \textit{Holothuria
  impatiens} complex. To this end, we delineated species within the complex
using (1) a threshold approach on a Neighbor-Joining tree, and (2) the general
mixed Yule-coalescent (GMYC) model on a tree obtained using Bayesian inference
with BEAST \cite{Drummond2012}.

To investigate the history of diversification and the affinities among the
cryptic species within the \textit{H. impatiens} complex, we built a
multi-locus, time-calibrated phylogeny.

Based on the putative species estimated by the genetic-based methods, we
investigated patterns of variation in the live appearance, ossicle shape,
habitat and geographical range to determine whether these additional lines of
evidence were concordant with the genetic data.

In some cases, genetic-based methods distinguished putative species that were
only slightly differentiated and could not be teased apart with other lines of
evidence. Therefore, we compared the marginal likelihoods of competing models of
species delineation using the multispecies coalescent in a Bayesian framework as
implemented in *BEAST \cite{Heled2010}.

\subsection{Sampling}

Specimens used for this study are housed in the collections of the Florida
Museum of Natural History (FLMNH) at the University of Florida, Gainesville, FL,
USA. A few additional tissue samples with voucher hosted in other institutions
or with no voucher were obtained through collaborators
(Table~\ref{tab:voucherInfo} for details).

Specimens were collected at low tide, on snorkel, on SCUBA or by dredging. Most
specimens were photographed while alive \textit{in situ} or in the lab,
anesthetized in a 1:1 solution of sea water and 7.5\% solution of magnesium
chloride hexahydrate, before fixation in 75\% ethanol. When possible tentacles
were clipped, immediately put in 95-99\% ethanol, and later used for DNA
extractions. These tissues are stored in the Genetic Resources Respository of
the FLMNH.

Two hundred and fifty three specimens were examined morphologically and 216 were
used for molecular analyses. These specimens were collected across the entire
known range of \textit{H. impatiens}: Mediterranean Sea, Carribean Sea, Red Sea,
Tropical Indian and Pacific Oceans.

% TODO: make specimen table

\subsection{DNA extraction and amplification}

DNA was extracted using either Invitrogen\texttrademark\ DNAZol\textregistered\
or Omega Bio-Tek\texttrademark\ E.Z.N.A\textregistered\ Mollusc DNA kit following
manufacturer recommendations. DNA was most often extracted from tentacles,
sometimes from gonads, longitudinal muscles or body wall. When possible, the
extractions were performed on tissue sampled in the field.

In this study we amplified the mitochondrial markers COI, 16S, ATP6 and the
nuclear markers histone 3 (H3a), 18S, ITS1-5.8S-ITS2, c0036 and c0775. ATP6
primers have been developed in this study based on sequences available for this
gene in GenBank. c0036 and c0775 markers have been developed from a 454 run on
genomic DNA from \textit{H. edulis}, and additional details will be provided in
a future study (\cite{MichonneauInPrep}). c0036 is a portion of the gene
encoding for an histone H3-like centromeric protein A-like, while c0775 is a
portion of the gene encoding for the protein SFI1 based on annotations of the
sea urchin \textit{Strongylocentrotus purpuratus} genome.

Primers and PCR conditions used are provided in
Table~\ref{tab:pcrConditions}. Because of the length of ITS1-5.8S-ITS2, we used
the sequencing primer fm-5.8S-f. The primers fm-ITS-f is the reverse complement
sequence of the primer 18S-1708R (\ref{}) and fm-ITS-r is the reverse complement
of LSUFW1 (\ref{}).

We conducted PCR in 25$\mu$L reactions using either 15.4$\mu$L of water,
2.5$\mu$L of Sigma-Alrich \textregistered 10X PCR buffer, 2.5$\mu$L of dNTP,
2$\mu$L of $MgCl_2$, 1$\mu$L of the forward primer, 1$\mu$L of the reverse
primer, and 0.1$\mu$L of Sigma-Aldrich\textregistered Jumpstart\texttrademark
Taq DNA polymerase or the Promega GoTaq MasterMix following manufacturer
recommendations.

Sequencing of PCR products was performed by the Interdisciplinary Center for
Biotechnology Research at the University of Florida. Chromatrograms were edited
in Geneious \ref{}.

\begin{table}
  \caption{Primers and PCR conditions for the markers used in this study.}
\begin{tabular}{ l p{6cm} l l p{1.3cm}}
  \hline
  Marker & Primer Forward ($5'-3'$) \& Primer Reverse ($5'-3'$)  & Temp &  Cycles & Ref. \\ \hline
  COI    & COIceF ACT GCC CAC GCC CTA GTA ATG ATA TTT TTT ATG GTN ATG CC \newline
           COIceR TCG TGT GTC TAC GTC CAT TCC TAC TGT RAA CAT RTG & 42 & 40 & \cite{Hoareau2010} \\
  16S    & 16SAR CGC CTG TTT ATC AAA AAC AT \newline 
           16SBR GCC GGT CTG AAC TCA GAT CAC GT & 52 & 35 & \cite{Arndt1996} \\
  ATP6   & ATP6f GGA CAA TTT TCC CCA GAC CT \newline 
           ATP6r GGT GAA GAG GGT GTT GAT GG & 42 & 40 & this study \\
  28S    & LSUFW1 AGC GGA GGA AAA GAA ACT A \newline 
           LSUREV2 ACG ATC GAT TTG CAC GTC AG & 42 & 40 & \cite{} \\
  ITS1-5.8S-ITS2 & fm-ITS-f AGG TGA ACC TGC AGA TGG ATC A \newline 
                   fm-5.8S-f CGT CGA TGA AGA ACG CAG YW \newline 
                   fm-ITS-r TAG TTT CTT TTC CTC CGC T & 45 & 40 & This study, \cite{}, \cite{} \\
  c0036  & c0036f TAA CGA CGG ATC TCA CGG AG \newline 
           c0036r AAT AAT GCT GGC GTG ACG TC & 42 & 45 & This study \\
  c0775  & c0775f GCT CTT CGT TCA ATT TAT CTC GC \newline
           c0775r GGG ATG CAG TTT GTC GAG TG & 42 & 40 &This study \\
  \hline
\end{tabular}
\label{tab:pcrConditions}
\end{table}

\subsection{Morphological analysis}

To compare intra- and inter-specific variation in ossicle size in the dorsal
body wall, we measured various characteristics of buttons and tables composing
the ossicle assemblage of \textit{Holothuria impatiens}
(Fig.\ref{fig:ossicleMeasurements}).

For 3-5 individuals of each ESU, we measured characteristics of at least 10
buttons and 10 tables. Measurements were made from  photographs of the
ossicles at 400x magnification using ImageJ (\ref{ImageJ}).

Measurements were analyzed in a principal component analysis in R
\ref{Rcoreteam}.

\subsection{Phylogeny and divergence time estimation}

% Talk about how to deal with ambiguities

Sequences obtained for each marker were aligned using MUSCLE \ref{} with the
default settings. All markers led to unambiguous alignments except ITS for which
we used Gblocks \ref{} to remove parts that were difficult to align using a
minimum length of a block of 5 (-b4=5) and all gap positions can be selected
(-b5=a).

We estimated the phylogenetic relationships and the timing of the divergence
among the different lineages of \textit{Holothuria impatiens} using BEAST v2.0.2
\cite{}. We used the closure of the isthmus of Panama to calibrate our
phylogeny. However, given that the sister ESU to the West Atlantic ESU is found
in the Galapagos and not in the Eastern Pacific, we used a log-normal prior
distribution on the stem node of these ESUs with a mean of 1.5 and a standard
deviation of 1 which corresponds to a mean of 4.482 My (0.865 -- 23.22 My 95\%
confidence interval) in real space.

We used PartitionFinder \cite{} to determine the best-fit partition scheme and
model of molecular of evolution for our dataset. We defined the data blocks such
that the three codon position for ATP6, COI, H3a, c0036 and c0775, and each of
16S, ITS and LSU were individual partitions. We used the greedy algorithm, with
linked branch lengths, no user tree, and models of molecular evolution available
by default in BEAST (JC69, HKY, GTR, without or with proportion of invariant
sites and/or Gamma distributed rates across sites). We selected the best-fit
models and partition schemes based on the Bayesian Information Criterion. The
best fit model had 9 partitions. After some initial test runs in BEAST, we
modified slightly the model of molecular evolution to improve mixing of the
MCMC, and reduced to the number of partitions to see Table \ref{tabPartitions}.

The analysis was run using independent strict molecular clock for each
locus. The tree prior was set to a Yule process and a random starting tree. The
Markov Chains were run for $1.58 . 10^8$ generations. The analysis was repeated
twice.

We used Tracer \cite{} to check that the MCMC chains had reach stationarity and
that the two independent runs were consistent. We ensured that enough of the posterior
distributions for each parameter was sampled by checking that the ESS values for
all parameters were above 200. After removing a 10\% burnin for each independent
run, the trees sampled were combined.


\subsection{GMYC}
To delineate putative species based on molecular data, we fit generalized mixed
Yule coalescent (GMYC) model (\cite{Pons2006}, \cite{Monaghan2009a},
\cite{Fujisawa2013}) to the COI sequence data. This approach attempts to detect
transitions in the branching rate of an ultrametric tree corresponding to the
expected increase in lineage accumulation resulting from the shift to speciation
to intraspecific coalescent events. We fit the single-threshold
(\cite{Pons2006}) GYMC model, as it was shown to perform as well or better than
the multiple-threshold model \cite{Fujisawa2013}. To determine the location of
the threshold, the GMYC model is fit at different nodes along the tree, and the
time that provides the best likelihood for the location of the switch is
selected. Clusters of sequences found beyond this threshold are considered to
represent different putative species. Singletons (unique sequences that do not
belong to one of the identified clusters) were not considered putative species
for the purpose of this study as mutliple individuals are needed to estimate the
parameters of the multispecies coalescent model \cite{Heled2010}.

The GMYC model was fit onto a the COI lineage genealogy estimated using BEAST
1.7.5. We enforced a strict molecular clock, specified a Yule diversification
process, a HKY+I+G model of molecular evolution, and ran the chains for $10^{8}$
generations.


% Method in Pons+2006 and 
% Examples: Volger+2008 (COTS paper), Marshall+2011 (cicadas in New Zealand);
% Henrich+2010 (water beetles in Australia)
% See: Esselstyn+2012 as it includes simulation study of statistical power of
% this approach; Lohse2009 for a critic

\subsection{Species delineation}

Given the shallow genetic divergence observed among some of the ESUs detected by
GYMC and the lack of morphological differentiation among them. We attempted to
compare species delineation models with *BEAST v 1.7.5 \cite{Heled2010} using
Bayes Factors evaluated by the stepping-stone (SS) and the path sampling (PS)
methods \cite{Baele2012}.

*BEAST infers the species trees from a multi-locus dataset by incorporating the
multi-species coalescent in a Bayesian framework to account for intraspecies
polymorphism and incomplete lineage sorting. This method assumes reproductive
isolation (absence of gene flow) among species and requires to assign
individuals to species \textit{a priori}.

Bayes Factors (BF) provides a powerful framework for model selection by
comparing marginal likelihoods. The models do not have to be nested and the
complexity of the models is directly accounted for by the marginal likelihoods
\cite{Baele2012}. Kass \& Raftery \cite{Kass1995} developed guidelines to
interpret BF values such that $2.ln(B_{01})$ (where $B_{01} =
\frac{p(Y|M_1)}{p(Y|M_0)}$ with $Y$ is the observed data and $M_i$ the model
under consideration) must be $> 6$ to consider the evidence ``strong'' and $>
10$ to consider it ``very strong''. Here, we use PS and SS methods to
estimate and compare the marginal likelihoods of alternative assignments of
individuals to species in *BEAST. With this approach, we test whether we can
detect signal of reproductive isolation among individuals that cannot be teased
apart reliably based on morphology alone.

We investigated whether *BEAST favored models where all or some of the following
ESUs were considered distinct species: WPac, Hawaii and RedSea or actually could
not be distinguished from ESU1 (Table \ref{tab:delineationModels}). We used
ESU3 and gracilis as outgroups. In these analyses, we selected individuals for
which we had at least one mitochondrial marker and one nuclear marker. We used
data from the following markers: 16S, COI, ATP6, c0036, c0775, ITS, LSU and H3a;
and the same approach as for the phylogeny of the entire complex, using Gblocks
to remove ambiguous parts of the alignment in ITS, PartitionFinder to find the
most appropriate partition scheme and models of molecular evolution. We unlinked
clocks for all markers, used strict molecular clocks, fixing the COI clock at
3.7\% \cite{Lessios2008} while estimating the rate of the others. We ran the
analysis for $5.10^{8}$ generations and repeated it twice for each gro.

For the estimation of the marginal likelihoods, we used the settings recommended
by Baele et al. \cite{Baele2012} with chain lengths of $10^{6}$, path steps of
100, and sampling along evenly spaced quantiles of a $Beta(0.3, 1)$ distribution.

\begin{table}
  \centering
  \caption{Species groupings used in models of species delineation.}
  \begin{tabular}{ l l l }
    \hline
    Model & Number of species & ESUs attributed to ESU1 \\
    \hline 
    M1    &     2             &  WPac, Hawaii, RedSea   \\
    M2    &     3             &  WPac, RedSea           \\
    M3    &     3             &  WPac, Hawaii           \\
    M4    &     4             &  WPac                   \\
    M5    &     4             &  Hawaii                 \\
    M6    &     4             &  RedSea                 \\
    M7    &     5             &  --                     \\
    \hline
  \end{tabular}
  \label{tab:delineationModels}
\end{table}


To confirm that the differentiation in COI which delineate the putative species
using GMYC does not drive the marginal likelihood calculations, and therefore
the model of species delineation favored by *BEAST, we repeated the analyses
omitting the mitochondrial markers.

\section{Results}

\subsection{Diversity}
% how many clades
% how many specimens per clades


\subsection{Phylogenetic relationships among the ESUs}
% TODO -- add definition of abbreviation for each ESU in figure legend if needed.
<<impatiens-tree, dependson='init-phylo', eval=TRUE, cache=TRUE, echo=FALSE, fig.keep='last', fig.height=8, fig.cap="Bayesian phylogenetic tree of the \\textit{Holothuria impatiens} complex. Vertical bars along the tips of the tree show the different ESUs found in the complex. Axis on the bottom in My.">>=
@

The \textit{Holothuria impatiens} includes 12, reciprocally monophyletic,
well-supported (all with posterior probabilities higher than 0.975), ESUs
(Fig~\ref{fig:impatiens-tree}). Some of these ESUs have restricted geographical
distributions to an oceanic bassin (Mediterranean, Caribbean, Eastern Pacific,
Red Sea) or to an archipelago (Hawaii and Galapagos), while other ESUs are wide
ranging and are found across the Indo-Pacific (ESU1, ESU2, and ESU3)
(Fig~\ref{fig:impatiens-map-WA}).

<<impatiens-map-WA, dependson='init-map', eval=TRUE, cache=TRUE, echo=FALSE, fig.keep="last", message=FALSE, warning=FALSE, results="hide", fig.cap="Distribution maps of the ESUs.">>=
@ 

<<impatiens-map-group1, dependson='init-map', eval=TRUE, cache=TRUE, echo=FALSE, fig.keep="last", message=FALSE, warning=FALSE, results="hide", fig.cap="Distribution map for the ESUs in group 1.">>=
@ 

<<impatiens-map-group2, dependson='init-map', eval=TRUE, cache=TRUE, echo=FALSE, fig.keep="last", message=FALSE, warning=FALSE, results="hide", fig.cap="Distribution map for the ESUs in group 2.">>=
@ 

\subsection{Species limits}




% cryptic species because indistinguishable, indistinguishable because similar
% looking or too much variation to make sense of it

% information about their biology gathered on the supposition it's a single
% species throughout its range

% from Volger+2012, most accurate COI divergence rate for echinoderms,
% 3.7(+/-)0.8%/Myr, from Lessios paper about Panamanian isthmus.

% look into splitstree (already installed into ~/Software folder), this allows
% to test for the robustness of signal in minimum spanning tree (see Volger+2012
% for details)

% look into \phi_{ST}, Fu's F, Tajima's D (reading the Arlequin manual)

% one of the reason, H impatiens has not been recognized as a species complex
% earlier, is due to the fact that some ESUs are widespread and others are
% overlapping. One of the reasons, these ESUs can overlap is that there was an
% ancestral widespread form which gave rise to the other species by parapatric
% speciation (e.g., as in Machel's hermit crabs), the alternative view is that
% each species recently extended their ranges. Should I try to use Lagrange? or
% Beast to reconstruct the phylogeographic history of the species to tease apart
% between these 2 hypotheses?
% to identify biogeographic regions, see if there are consistent phylogeographic
% patterns among ESUs

\section{Discussion}

% According to \cite{Renema2008}, the modern IWP fauna has its origin in the
% Miocene. Generation of hotspots associated with tectonic events, so importance
% of abiotic factors.

% Look into \cite{Lessios1999} to see how the timing in Eucidaris compares to
% the one H. impatiens

% Therefore, as it appears that the majority of speciation events are
% allopatric (\cite{Coyne2004}), recent changes in distribution and population
% sizes must be prevalent in the \textit{H. impatiens} complex to explain current
% distribution patterns. These changes have implications regarding the rapid
% evolution of reprodutive isolation barriers and ecological adaptations.

% Add something about how differences in colors among ESUs might be a result of
% the same type of proteins than GRP.

% Low PP on some nodes, probably from using low number of loci. This low number
% of loci, is also not enough to obtain accurate population estimates (see
% Harris+2013,10.1111/evo.12280).

\section{Acknowledgments}

CIPRES portal.
UF HPC, HiPerGator.

\bibliographystyle{plos2009}
\bibliography{impatiens_phylogeography}

\begin{table}
  \centering
  \begin{tabular} { l l l }
    1      & $GTR+I+G$    & $16S, ATP6_2$ \\
    2      & $HKY+I+G$    & $ATP6_1, COI_1$\\ 
    3      & $HKY+I+G$    & $ATP6_3, c0775_1, c0775_2, c0775_3$\\ 
    4      & $GTR+I+G$    & $c0036_1, c0036_2, c0036_3$\\ 
    5      & $GTR+I+G$    & $COI_2, H3a_2$ \\
    6      & $HKY+I$      & $COI_3$\\ 
    7      & $HKY$        & $H3a_1, LSU$ \\ 
    8      & $JC$         & $H3a_3$ \\
    8      & $HKY+G$      & ITS \\       
  \end{tabular}
\end{table}




%% Notes about the biology of the species
%% - EP: discharge Cuv tubules \cite{Bakus1974}, ``relatively very active''
%% \cite{Bakus1974}, 
%% - Egypt: tables appear earlier than buttons in juveniles, look for pl.8 fig4-5,
%% and pl.9 2 in Savigny \cite{Mortensen1902}10.1111/j.1096-3642.1926.tb00326.x

%% READ \cite{Cutress1996}, seems like there is a lot of information about
%% ontogenetic changes in the shape of the ossicles.
%% ``Buttons in OM dermis of the 195 mm specimen from Ceylon, however, are
%% distinctly smaller.'' 

\end{document}




