% -*- 
\documentclass{article}

\usepackage{authblk}
\usepackage{tabularx}
\usepackage{float}
\usepackage{amsmath}
\usepackage[margin=1in]{geometry}
\usepackage[font=sf]{caption}

% Latex special characters are rendered correctly with XeTeX
\usepackage{xltxtra}
\usepackage{xunicode}
\defaultfontfeatures{Mapping=tex-text}

% Words are cut where needed
\usepackage{polyglossia}
\setdefaultlanguage[variant=american]{english}

% Use fancy fonts
\usepackage{fontspec}
\setmainfont{Crimson Text}
\setsansfont{SourceSansPro-Regular}

% tables
\usepackage{multirow}

\newcolumntype{b}{X}
\newcolumntype{s}{>{\hsize=.2\hsize}X}

\title{Cryptic and not-so-cryptic species in the complex ``\textit{Holothuria
    (Thymiosycia) imaptiens}'' (Forrsk\r{a}ll, 1775) (Echinodermata:
  Holothuroidea: Holothuriidae)} \author[1]{Fran\c{c}ois
  Michonneau\thanks{Corresponding author: francois.michonneau@gmail.com}}
\author[1]{Gustav Paulay} \affil[1]{Florida Museum of Natural History,
  University of Florida, Gainesville, FL 32611-7800, USA}

\renewcommand\Authands{ and }


\begin{document}

<<setup, echo=FALSE, cache=FALSE, include=FALSE>>=
opts_chunk$set(fig.path='figures/latex-', cache.path='cache/latex-', dev="tikz",
               external=TRUE, fig.width=7, fig.height=4, fig.align="center")
##knit_hooks$set(crop = hook_pdfcrop)
options(tikzDefaultEngine = "xetex")
options(tikzDocumentDeclaration = paste(getOption("tikzDocumentDeclaration"),
            "\n\\renewcommand*{\\familydefault}{\\sfdefault}", sep=""))
read_chunk('code/impatiens_analysis.R')
read_chunk('code/impatiens_map.R')
@ 

<<init-phylo, echo=FALSE, cache=TRUE, include=FALSE>>=
@ 

<<init-map, echo=FALSE, cache=TRUE, include=FALSE>>=
@ 

\maketitle

\begin{abstract}
Blabla
\end{abstract}

\section{Introduction}

Despite the challenges associated with its definition, species are, by far, the
most commonly used biological unit to assess patterns of diversity, identify
regions of conservation concern and manage exploited resources. For many
taxonomic groups, identifying species based on morphological characters alone
can be problematic as species limits might be misunderstood. Barcoding, the
molecular-based identification methods that use a short DNA fragment (typically
mitochondrial DNA in metazoans) provide an inexpensive and efficient way to
identify species complexes, and gain insights into species limits. These methods
are efficient as genetic data can carry the signature of the historical
processes that have shaped modern patterns. In particular, the expected level of
genetic divergence between reproductively isolated populations is proportional
to the amount of time since they shared a common ancestor
\cite{Hey2010-IMa2}. However, divergence in mitochondrial DNA alone is
insufficient to conclude for reproductive isolation. Historical interruption of
gene flow between populations might lead to genetic divergence, but without
reproductive isolation, upon secondary contact, these populations will resume
inter-breeding. Conversely, introgression \cite{} and incomplete lineage sorting
\cite{} can mask reproductive isolation.

% use framework put together for NAEC/Evolution for introduction

% include something on how importance of rate of secondary sympatry for building
% up local diversity, and confusion about sympatry + genetic differentiation in
% case of no clear barriers and possibility for rapid changes in population
% sizes, geographical ranges

Based on the precept that species are defined on the criterion of reproductive
isolation, three processes operating at different rates are interacting during
speciation: (1) acquisition of reproductive isolation, (2) morphological
differentiation, (3) genetic differentiation. In many cases, these three
processes can be independent making species delineation challenging. For
instance, broadcast spawning marine invertebrates for which reproductive
cohesion is maintained behaviorally (e.g., spawning aggregations cukes?,
lobsters? \cite{}, spawning based on environmental clues corals? \cite{}) and
biochemically (e.g., gamete recognition proteins \cite{}), selection on these
traits is unlikely to drive morphological differentiation among closely related
species \cite{Knowlton1993}. Similarly, barcoding markers evolve typically
neutrally and the time taken for them to show enough differentiation to be
recognized as different species will depend on effective population sizes or DNA
repair mechanisms (in absence of introgression) that will also be independent of
the species identity.

Assessing divergence in characters used by species to maintain reproductive
isolation is challenging as they are in most cases unknown or poorly
understood. We therefore only have access to morphological and genetic
information for which the differentiation is a by-product of reproductive
isolation that do not guarantee to be effective in discriminating
species. Integrative taxonomy has been proposed to integrate these multiple
lines of evidence ...

The failure to distinguish reproductively isolated species can stem from either
the choice of an inappropriate set of characters to distinguish species
otherwise morphologically variable, or from the absence of variation in
morphology but differentiation in other traits (e.g., ecology, chemical cues)
allows them to remain reproductively isolated

For morphological characters to be appropriate to distinguish between
populations of reproductively isolated individuals, they require to evolve at a
rate of the same par as populations become species. They also must exhibit less
intraspecific than interspecific variation. Traits that are under sexual
selection usually meet these criteria. When species use visual clues to identify
their mates, taxonomists can use the same characters to discrimminate between
these populations of reproductively isolated individuals. However, in situations
where mate recognition does not involve visual clues (e.g., chemical
communication, sperm/egg recognition proteins),

Most marine invertebrate taxonomy results from the study of dry or preserved
specimens. With these methods, many delicate morphological characters and very
often color patterns are lost. Therefore, in groups characterized by hard parts,
taxonomists have relied on these skeletal elements to find characters that allow
them to differentiate species. For instance, in sea cucumbers, in particular in
the Holothuriidae, species-level taxonomy has entirely been focused on the shape
of ossicles, microscopic calcareous secretions found in many tissues. At least
until the late 1980's most species descriptions would ignore intraindividual
variation by figuring only ossicles found in the body wall of the species, and
generally ignored any intraspecific variation by illustrating the ossicle
assemblage of a single individual.

% who were the first authors to illustrate ossicles?

Forrsk\r{a}ll described \textit{Holothuria impatiens} in 1775 from the
collections he was making in Suez, Egypt. The original description is vague, and
could fit many other holothuriid species of the area, he did however include a
drawing of the species. Since then, the range of \textit{H. impatiens} has been
extended, and the species is now thought to be occurring in the Red Sea, the
entire Indo-West Pacific region, the Caribbean and the Mediterrean seas.

\textit{Holothuria impatiens} is the type species of the subgenus
\textit{Thymiosycia} one of the 18 subgenera that are forming the genus
\textit{Holothuria}. The subgenus \textit{Thymiosycia} was first erected by
Pearson \cite{Pearson1915} to include all holothuriid being characterized by the
absence of ``true pedicels'' and anal teeth, and the presence of ``papillae
scattered all over the body, generally on eminences''; and designed
\textit{H. impatiens} as the type species for the subgenus. In the most recent
revision of the family, Rowe \cite{Rowe1969} redefined \textit{Thymiosycia} to
only include species which are characterized by having well developed, regular
and symmetrical tables and buttons. The ossicle assemblage of
\textit{H. impatiens} therefore exemplifies the holothuriid ossicles.

\textit{Holothuria impatiens} is a common, locally fairly abundant, species
which can be found under rocks in shallow waters (in particular in lagoons and
back-reef habitats) but has been recorded down to 158~m (\cite{Samyn2014-Miriky}).
Because of its ubiquituous distribution, it is one of the more studied species
of holothuriid with studies on its reproduction \cite{Harriott1985}, its
Cuvierian tubules \cite{Flammang2002}, \cite{Becker+Flammang2010}; its toxicity
\cite{Bakus1974}; its feeding preferences \cite{Roberts1982}; the chemical
composition \cite{Hampton1958}, statistical analysis \cite{Hampton1959},
ontogenic changes in its ossicles \cite{Cutress1996}; and its parasites
\cite{Martens1994}.  It has also been included in molecular phylogenies (e.g.,
\cite{Lacey2005}, \cite{Honey-Escandon2012}). \textit{Holothuria impatiens} is a
low-value commercial species but is reported to be fished in the Eastern Pacific
\cite{Toral-Granda2008}, Madagascar \cite{Conand2007}, and Palau
\cite{Pakoa2009}.

Despite its relative biological and commercial importance, the variation
observed in their color patterns and reported by previous workers (e.g.,
\cite[p.178]{Clark1921}, \cite{Rowe+Richmond2004}) has yet to be
investigated. In the present study, using a combination of morphological,
genetic and ecological data, we show that \textit{Holothuria impatiens}
represents a complex of at least 12 putative species. If some of these species
can be distinguished relatively easily from their live appearance, others are
harder to tease apart. This, in combination with the fact that several of these
species co-occur, and sometimes even share the same habitat, has contributed to
the failure to identify the species limits in this complex.

The development of genetic data and its application to species identification
have allowed the discovery of many species complexes. The DNA barcoding approach
of using an easy-to-amplify single locus to delineate species provides a
promising venue to speed up biodiversity documentation. Ideally the analysis of
the genetic data should delineate the biological species accurately with little
or no subjectivity, so that the process can be automated and used on large
datasets. However the biological processes generating the genetic
differentiation that can be used to delineate species, depends on a variety of
other factors than reproductive isolation that a single locus cannot
capture. Nonetheless, threshold based methods (e.g. \cite{}) or other methods
developped \cite{Pons2006}, \cite{Monaghan2009} to detect species limits are
relatively succesful. While the possibility of detecting accurately species
limits from a single marker remains contentious, using multiple lines of
evidence to assess species limits in species complexes detected from
single-locus dataset is widespread. The development of integrative taxonomy
which uses multiple lines of evidence to delinate species has recently seen more
formal development in how to determine the species limits based on available
evidence. The most recent developments favor finding a biological explanation
behind contradictory lines of evidence \cite{Schlick-Steiner2010}. For genetic
data, recent developments in Bayesian statistics allow for integrating genetic
signal from a variety of markers by invoking the multi-coalescent process and
also compare competing models.

The genetic differentiation observed in species previously thought to be found
across large geographic areas (e.g, circum-tropical species) has changed our
understanding of how biodiversity is distributed in the sea, with isolated
oceanic bassins and archipelagoes exhibiting much higher endemism rates than
previously thought (e.g., %find references for Caribbean, Red Sea, Medit
\cite{Meyer2005a}). If unraveling diversity among geographically isolated
populations affects estimates of global diversity, unraveling diversity among
sympatric populations affects not only local estimates of diversity, but also
challenges our understanding of the processes generating biodiversity.

Understanding the dynamics of diversification in the sea is challenged by the
fact that most marine invertebrates have a highly dispersive larval stage and
that oceans in general and the Indo-Pacific in particular do not seem to have
clear geographical barriers to dispersal. Yet, while some species have broad
geographical ranges encompassing most of the IWP province, others are more
limited, and some closely related species have allopatric distributions. The
frequent overlap in species distribution across the IWP challenges our current
understanding of the role of geographical isolation in speciation. It is
generally accepted, at least on land, that allopatry is the most common mode of
speciation. However, the combination of broadly overlapping ranges and high
diversity leaves seemingly few opportunities for allopatry in the
sea. Furthermore, recent geological and molecular evidence suggests that most of
the modern reef fauna was shaped during the Miocene. To account for this
apparent paradox, several alternative models have been proposed. One category
downplays the importance of the absence of gene flow (allopatry) and emphasizes
the role of selection (along environemental gradients, sexual selection, host
specificity) as the main driver behind reproductive isolation (see
\cite{Bowen2013}). Another category of models proposes that speciation occurred
in allopatry, but since then, important changes in species ranges and
distributions have obscurred the barriers responsible for the initiation of the
reproductive isolation.

Inferring the mode of speciation from current distributions in the Indo-Pacific
is most likely misleading as modern ranges in the Pacific are necessarily
relatively recent as sea levels have fluctuated greatly (ca. 100-150~m) and
repeatedly (ca. 100~k.y. periodicity) during the late Tertiary and early
Quarternary. These fluctuations have led to important habitat modifications and
changes in oceanic current patterns with consequences in connectivity,
population sizes and species ranges. These changes have made species
distributions in the Pacific highly dynamic (\cite{Paulay1990}) and are, in most
cases, too recent compared to the time frame needed for speciation. 
% However, speciation can be very rapid, e.g., Asterinid from NSW Byrne2012
For instance, the latest minimum sea level (120 to 135~m below present) was
18,000 years ago. During this period, circulation between Pacific and Indian
Oceans was significantly restricted, the Red Sea was most likely disconnected
from the Indian Ocean, and the South China sea was land-locked
(\cite{Veron1995}). Therefore, current distribution patterns reflect only
partially the biogeographical context under which the diversification of modern
reef-associated fauna has occurred.

The consequences of these fluctuations in sea level on the populations might
however still be visible in the present genetic diversity of the modern
fauna. For instance, the low sea levels caused an important reduction of the
connectivity between the Indian and Pacific oceans providing repeated
opportunity for population divergence between the two ocean bassins. While some
species retained a genetic signature of this episode of low connectivity in
their mitochondrial DNA (e.g., xxxxxx), many other lineages (including the most
widespread form of \textit{Holothuria impatiens}) show little geographic genetic
structuring across their range. This latter pattern can be explained by (1)
maintained connectivity during these low sea stands, as both oceans were still
connected (\cite{Voris2000}); (2) one of the two lineages was lost because of
local extinction or selective sweep; (3) the species recently expanded their
ranges and colonized one of the bassin very recently.

In this study, we use comprehensive sampling in the complex \textit{Holothuria
  impatiens} covering its entire known geographic range, to investigate species
distribution and patterns of speciation. Specifically, we aim to:
\begin{itemize}
  \item assess species limits in the complex;
  \item determine morphological characters that can be used to distinguish among
    the species;
  \item reconstruct the diversification history of the complex to understand how
    species came to co-occur.
\end{itemize}
We compare patterns of species distribution with a time-calibrated phylogeny,
and test species limits in a subset of the species that are not morphologically
differentiated using the multi-species coalescent statistical framework. We also
investigate how the general mixed Yule coalescent model of species delineation
performs with our data set, and how sensitive this method is to the parameters
used to reconstruct the phylogeny.

% Allopatry without clear geographic barriers (Diadema setosum, Lessios 2001)

% how to include peripatric speciation?
% say something about how these 2 models are not mutually exclusive

% use these bayesian methods and test for adequacy of multispecies coalescent
% justified because wide ranging species with no clear geographical barriers to
% their range.

\section{Methods}

\subsection{Approach}

The goal of this study is to identify cryptic species in the complex
\textit{Holothuria impatiens}, and to understand the temporal and spatial
dynamics of its diversification.

We use a combination of complementary methods of species delineation following
modifications of the approaches by \cite{Leache2010,Satler2013}. First, we used
the Generalized Mixed Yule Coalescent method (GMYC) to identify putative species
\cite{Pons2006,Monaghan2009}. We then used independent lines of evidence
(morphology, ecology and geographic information) to assess the validity of the
lineages identified with GMYC. In the event that no line of evidence could
separate the lineages identified with GMYC, we used model comparison in *BEAST
\cite{Heled2010,Baele2012} to assign species limits appropriately.

We use this approach as the complex includes both deeply divergent lineages that
can readily be distinguished based on morphological characters, and recent
radiations that are morphologically indistinguishable. The multispecies
coalescent offers a statistical framework that provides an objective assessment
of species limits \cite{Fujita2012a}, even when no other morphological,
ecological or geographical information allows to identify independent
evolutionary lineages. Our study shows these methods can be added to the lines
of evidence typically used in integrative taxonomy.

Additionally, to investigate the history of diversification and the affinities
among the cryptic species within the \textit{H. impatiens} complex, we built a
multi-locus, time-calibrated phylogeny.

\subsection{Sampling}

Specimens used for this study are housed in the Invertebrate Zoology collections
of the Florida Museum of Natural History (FLMNH) at the University of Florida,
Gainesville, FL, USA. A few additional tissue samples with voucher hosted in
other institutions or with no voucher were obtained through collaborators
(Table~\ref{tab:voucherInfo} for details).

Specimens were collected at low tide, on snorkel, on SCUBA or by dredging. Most
specimens were photographed while alive \textit{in situ} or in the lab,
anesthetized in a 1:1 solution of sea water and 7.5\% solution of magnesium
chloride hexahydrate, before fixation in 75\% ethanol. When possible tentacles
were clipped, immediately put in 95-99\% ethanol, and later used for DNA
extractions. These tissues are stored in the Genetic Resources Respository of
the FLMNH.

Two hundred and fifty three specimens were examined morphologically and 216 were
used for molecular analyses. These specimens were collected across the entire
known range of \textit{H. impatiens}: Mediterranean Sea, Carribean Sea, Red Sea,
tropical Indian and Pacific Oceans.

% TODO: make specimen table

\subsection{DNA extraction and amplification}

<<number-ambiguities, echo=FALSE, cache=TRUE>>=
@ 

DNA was extracted using either Invitrogen\texttrademark\ DNAZol\textregistered\
or Omega Bio-Tek\texttrademark\ E.Z.N.A\textregistered\ Mollusc DNA kit following
manufacturer recommendations. DNA was most often extracted from tentacles,
sometimes from gonads, longitudinal muscles or body wall. When possible, the
extractions were performed on tissue sampled in the field.

In this study we amplified the mitochondrial markers COI, 16S, ATP6 and the
nuclear markers histone 3 (H3a), 18S, ITS1-5.8S-ITS2, c0036 and c0775. ATP6
primers have been developed in this study based on sequences available for this
gene in GenBank. c0036 and c0775 are anonymous markers developed from a 454 run
on genomic DNA from \textit{H. edulis}, and additional details will be provided
in a future study (\cite{MichonneauInPrep}). c0036 is a portion of the gene
encoding for an histone H3-like centromeric protein A-like (locus XP\_003723879),
while c0775 is a portion of the gene encoding for the protein SFI1 based on
annotations of the sea urchin \textit{Strongylocentrotus purpuratus} genome
(locus XP\_792620).

Primers and PCR conditions used are provided in
Table~\ref{tab:pcrConditions}. Because of the length of ITS1-5.8S-ITS2, we used
the sequencing primer fm-5.8S-f. The primers fm-ITS-f is the reverse complement
sequence of the primer 18S-1708R (WN-1708R in \ref{Lacey2005}) and fm-ITS-r is
the reverse complement of LSUFW1 (LSU D1,D2 fw1: 56-74 in \ref{Sonnenberg2007}).

We conducted PCR in 25$\mu$L reactions using either 15.4$\mu$L of water,
2.5$\mu$L of Sigma-Alrich \textregistered 10X PCR buffer, 2.5$\mu$L of dNTP,
2$\mu$L of $MgCl_2$, 1$\mu$L of the forward primer, 1$\mu$L of the reverse
primer, and 0.1$\mu$L of Sigma-Aldrich\textregistered Jumpstart\texttrademark
Taq DNA polymerase or the Promega GoTaq MasterMix following manufacturer
recommendations.

Sequencing of PCR products was performed by the Interdisciplinary Center for
Biotechnology Research at the University of Florida. Chromatrograms were edited
using Geneious \ref{Geneious558}.

Overall, \Sexpr{nAmbTotal} positions (\Sexpr{round(nAmbPositions/dim(fullAlg)[2],
  1)*100}\% of the alignment length, and \Sexpr{round(nAmbTotal/totalNucl,
  4)*100}\% of the total number of nucleotides) contained heterozygous
positions. We kept the IUPAC in the alignment which are considered as missing
data by all the software used in this study.

\begin{table}
  \caption{Primers and PCR conditions for the markers used in this study.}
\begin{tabular}{ l l p{6cm} l l l}
  \hline
  Marker & Primer & Primer Forward ($5'-3'$) \& Primer Reverse ($5'-3'$)  & Temp &  Cycles & Ref. \\
  \hline
  % COI --- 
  COI &  COIceF  & ACT GCC CAC GCC CTA GTA ATG ATA TTT TTT ATG GTN ATG CC & 42 & 40 & \multirow{2}{*}{\cite{Hoareau2010}} \\
       & COIceR  & TCG TGT GTC TAC GTC CAT TCC TAC TGT RAA CAT RTG & & & \\
  % 16S ---
  16S & 16SAR & CGC CTG TTT ATC AAA AAC AT & 52 & 35 & \multirow{2}{*}{\cite{Arndt1996}} \\
      & 16SBR & GCC GGT CTG AAC TCA GAT CAC GT & & & \\
  % 16Sc ---
  %% TODO 
  % ATP6 ---
  ATP6 & ATP6f   & GGA CAA TTT TCC CCA GAC CT & 42 & 40 & \multirow{2}{*}{This study} \\
       & ATP6r   & GGT GAA GAG GGT GTT GAT GG & & & \\
  % 28S ---
  28S  & LSUFW1  & AGC GGA GGA AAA GAA ACT A &  42 & 40 & \multirow{2}{*}{\cite{Sonnenberg2007}} \\
       & LSUREV2 & ACG ATC GAT TTG CAC GTC AG & & & \\
  % ITS ---
  ITS1-5.8S-ITS2 & fm-ITS-f  & AGG TGA ACC TGC AGA TGG ATC A & 45 & 40 & \multirow{2}{*}{This study, \cite{Lacey2005,Sonnenberg2007}} \\
                  & fm-5.8S-f & CGT CGA TGA AGA ACG CAG YW & & & \\
                  & fm-ITS-r  & TAG TTT CTT TTC CTC CGC T  & & & \\
  % c0036 ---
  c0036 & c0036f & TAA CGA CGG ATC TCA CGG AG & 42 & 45 & \multirow{2}{*}{This study} \\
        & c0036r & AAT AAT GCT GGC GTG ACG TC & & & \\
  % c0775 ---
  c0775 & c0775f & GCT CTT CGT TCA ATT TAT CTC GC & 42 & 40 & \multirow{2}{*}{This study} \\
        & c0775r & GGG ATG CAG TTT GTC GAG TG & & & \\
  \hline
\end{tabular}
\label{tab:pcrConditions}
\end{table}

%% \subsection{Morphological analysis}

%% To compare intra- and inter-specific variation in ossicle size and shape in the
%% dorsal body wall, we measured various characteristics of buttons and tables
%% composing the ossicle assemblage of \textit{Holothuria impatiens}
%% (Fig.\ref{fig:ossicleMeasurements}).

%% For 3-5 individuals of each ESU, we measured characteristics of at least 10
%% buttons and 10 tables. Measurements were made from  photographs of the
%% ossicles at 400x magnification using ImageJ (\ref{ImageJ}).

%% Measurements were analyzed in a principal component analysis in R
%% \ref{Rcoreteam}.

\subsection{Putative species delineation with GMYC}

% Justify why using COI sequences only for this.

To delineate putative species based on molecular data, we fit the generalized
mixed Yule coalescent (GMYC) model (\cite{Pons2006}, \cite{Monaghan2009},
\cite{Fujisawa2013}) to the COI sequence data. This approach attempts to detect
transitions in the branching rate of an ultrametric tree corresponding to the
expected increase in lineage accumulation resulting from the shift between
speciation and population-level coalescent events. To determine the location of
the threshold, the GMYC model is fit at different nodes along the tree, and the
time that provides the best likelihood for the location of the switch is
selected. Lineages found beyond this threshold are considered different species.

Estimating the transition from speciation to intraspecific coalescent events
requires an ultrametric tree. We used BEAST to obtain these trees as it provides
a powerful statistical framework to interpret the chronological context of
molecular variation \cite{Drummond2014}. Despite some investigations of the
effects of priors on the number of species estimated by GMYC
(\cite{Monaghan2009, Fujisawa2013, Talavera2013}), we here revisit the impact of
specifying a strict or a log-normal relaxed clock, as well as the type of tree
prior used (coalescent with constant population size \cite{Kingman1982},
coalescent with exponentially growing population size \cite{Griffiths1994}, and
Yule \cite{Yule1925}). The type of clock used will affect branch lengths,
however, it is difficult to predict \textit{a priori} how the misspecification
of this parameter will influence the results. Monaghan et
al. \cite{Monaghan2009} indicate that using a coalescent constant tree prior is
a more conservative approach compared to a Yule prior, as GMYC uses the
coalescent as the null model to explain the branching pattern. Their results
validated their predictions and the GMYC method detected more species on trees
estimated with a Yule prior. However, these results might be influenced by the
nature of the data, in particular by the amount of variation observed in the
sequence data, and deserve further testing.

Furthermore, to our knowledge few studies have investigated the effect of
retaining identical haplotypes in the analysis (but see \cite{Talavera2013}). In
other studies, identical sequences were removed as they were thought to make the
task of estimating the transition problematic \cite{Monaghan2009}, as the GMYC
method cannot accommodate for the null branch lengths associated with identical
sequences \cite{Fujisawa2013}. Contrary to other programs, BEAST does not assign
zero edge length to the terminal branches for terminal sequences, and thus
allows to retain them in the analysis. There are obvious theoretical advantages
for retaining identical sequences: BEAST assumes that the sequences represent a
random sample of the population when estimating effective population
sizes. Removing identical sequences will lead to spurious high levels of genetic
diversity, in turn leading to overestimating population sizes
\cite{Drummond2014}[p.98]. As population sizes directly influence the
probability that lineages will coalesce, removing identical sequences will lead
to longer branches in the trees, blurring the distinction between interspecific
and intraspecific coalescent events. We can therefore predict that removing
identical sequences of the analysis will lead to higher number of species
detected by GMYC and higher uncertainty, both of which are undesirable for the
purpose of the analysis.

We performed the GMYC analyses using only COI sequences as we want to compare
the performance of this method with the results of the complete analysis. COI
genealogies were estimated using BEAST 1.8.0 \cite{Drummond2012} and BEAGLE 2.1
\cite{Ayres2012}. To determine the optimal partition scheme, we used
PartitionFinder \cite{Lanfear2012} on the dataset including all sequences and
the dataset including the haplotypes. We performed a complete search using each
codon position as a partition, and limiting the search to the JC, HKY and GTR
models with or without a proportion of invariant sites and/or gamma-distributed
rates of evolution (12 different models \textit{in toto}). Both datasets (all
sequences, only haplotypes) favored to assign each codon position to a different
partition. After inspecting mixing and convergence on test runs, we modified
slightly the models suggested by PartitionFinder to use: HKY+I+G, GTR+G and HKY
for for each codon position respectively for dataset including all sequences;
and HKY+G, HKY+G and HKY for analyses on unique haplotypes. In all analyses,
base frequencies were estimated.

We used a log-normal distribution as the prior for the mean rate of substitution
for the log-normal relaxed clock and the strict clock, with a \textit{log}(mean)
of -4 and a \textit{log}(standard deviation) of 1. We calibrated the COI
genealogies using the closure of the isthmus of Panama. We constrained the age
of the node corresponding to the most recent common ancestor of the individuals
found in the Galapagos archipelago, the coast of the Eastern Pacific and in the
Caribbean using a log-normal prior with a \textit{log}(mean) of 1.5, a
\textit{log}(standard deviation) of 0.75 and an offset of 2.5. This distribution
translates into a mean age of divergence set at \Sexpr{round(qlnorm(0.5,
  meanlog=1.5, sdlog=0.75) + 2.5, 2)} millions years (MY) ago, (95\% confidence
interval of [\Sexpr{paste0(round(qlnorm(c(0.025, 0.975), meanlog=1.5,
  sdlog=0.75) + 2.5, 2), collapse=", ")}]).

We set the MCMC chain lengths at 5.10$^{7}$, sampling every 5.10$^{3}$
generations. All analyses were ran twice from independent starting points and
were checked for convergence. From visual inspection of the values sampled by
the chains with Tracer 1.6, we determined that a burnin of 10\% was sufficient
for all runs. ESS values for all parameters were above 350 in each independent
run, but both runs for each parameter combination were nonetheless combined.

The trees sampled from the posterior were summarized with treeannotator using
the common ancestor tree method \cite{Heled2013} as it provides a more accurate
estimate of the divergence times from the trees sampled from the posterior.

These trees were used to fit the single-threshold (\cite{Pons2006}) and the
multi-threshold \cite{Monaghan2009} GYMC model using the R package splits 1.0-19
\cite{Ezard2013}. We considered the number of ``entities'' returned by GMYC as
the estimated number of species in our analyses. The confidence interval for the
number of species corresponds to the minimum and maximum numbers of species for
threshold values found within 2 log-likelihood units of the threshold associated
with the highest likelihood.


\subsection{Phylogeny and divergence time estimation}

<<median-tmrca-WAgroup, eval=TRUE, echo=FALSE, cache=TRUE, dependson='init-phylo'>>=
@ 

Sequences obtained for each marker were aligned using MUSCLE \cite{Edgar2004}
with the default settings. All markers led to unambiguous alignments except ITS
for which we used Gblocks \ref{Castresana2000} to remove parts that were
difficult to align using a minimum length of a block of 5 (-b4=5) and all gap
positions can be selected (-b5=a). We used the resulting alignment to estimate
the phylogenetic relationships and the timing of the divergence among the
different lineages of \textit{Holothuria impatiens} using RAxML 8.0.1
\cite{Stamatakis2014} and BEAST 2.1.2 \cite{Bouckaert2014} (using BEAGLE 2.1.2
\cite{Ayres2012}).

To determine the optimal partition scheme and models of molecular evolution, we
analyzed the alignment using PartitionFinder \cite{Lanfear2012}. We defined the
data blocks such that the three codon position for ATP6, COI, H3a, and each of
16S, c0036, c0775, ITS and LSU were individual partitions. We used the greedy
algorithm, with linked branch lengths, no user tree, and models of molecular
evolution available by default in BEAST (JC69, HKY, GTR, without or with a
proportion of invariant sites [+I] and/or Gamma distributed rates across sites
[+G]). We selected the best-fit models and partition schemes based on the
Bayesian Information Criterion. The best fit model had 9 partitions.

For the BEAST analysis, following some initial testing runs, we modified
slightly the partition scheme to improve mixing of the chains (supplementary
Table \ref{tab:sm-partitions}).

As no fossils are known in \textit{Holothuria}, we used the closure of the
isthmus of Panama to calibrate our phylogeny, and fixed the time of divergence
for the individuals of the Eastern Pacific (including Galapagos) and the
Caribbean to 5 MY. The isthmus of Panama was completely closed about 3 MY ago
but most species were probably isolated earlier \cite{Lessios2008}. We chose a
conservative estimate to account for the fact that the sister species lineages
found across the isthmus are the lineages found in the Galapagos and the
Caribbean (the coastal Eastern Pacific lineage is sister to the one found in the
Galapagos, Figure~\ref{impatiens-tree}), suggesting that the divergence might
have been preceded the closure of the isthmus.


% Additionally, the estimated age of this node across the COI genealogies used for
% the GMYC analyses, varied between \Sexpr{round(min(meanTmrca), 2)} and
% \Sexpr{round(max(meanTmrca), 2)} MY (estimated from the mean on the posterior
% distributions, prior \Sexpr{round(exp(1.5 + (.75^2)/2) + 2.5, 2)} MY; median
% ranged between \Sexpr{round(min(medianTmrca), 2)} and
% \Sexpr{round(max(medianTmrca), 2)} MY, prior
% \Sexpr{qlnorm(.5, meanlog=1.5, sdlog=.75) + 2.5} MY).

We used three independent strict molecular clocks (one for each of mitochondrial
markers, protein-coding nuclear markers, nuclear ribosomal markers)
\cite{Hasegawa1985}. The tree prior was set to a Yule process, and a random
starting tree. The Markov chains were run for $185.10^6$ generations (and
sampled every $5.10^3$ generations). The analysis was repeated twice from
independent starting points.

% We checked the operator file to ensure appropriate acceptance rates while
% sampling the posterior. %% no ops file in BEAST2?

We used Tracer 1.6 to check that the MCMC chains had reached stationarity, that
mixing was adequate, and that the two independent runs were consistent. We
ensured that a representative sample of the posterior distributions was sampled
for each independent runs, by checking that the ESS values for all parameters
were above 200. The samples from the posterior distribution from both runs were
combined using locombiner after removing a 20\% burnin, resulting in 29601
states sampled. The ESS values obtained by combining the runs were all above
440. We summarized the trees sampled with treeannotator using the common ancestor
tree method \cite{Heled2013}.

\subsection{Test of species limits with *BEAST}

Given the shallow genetic divergence observed among some of the ESUs detected by
GYMC and the lack of morphological differentiation among them, we compared
species delineation models with *BEAST v 1.7.5 \cite{Heled2010} using Bayes
Factors evaluated by the stepping-stone (SS) and the path sampling (PS) methods
\cite{Baele2012}.

*BEAST infers the species trees from a multi-locus dataset by incorporating the
multi-species coalescent in a Bayesian framework to account for intraspecies
polymorphism and incomplete lineage sorting. This method assumes reproductive
isolation (absence of gene flow) among species and requires to assign
individuals to species \textit{a priori}.

Bayes Factors (BF) provides a powerful framework for model selection by
comparing marginal likelihoods. The models do not have to be nested and the
complexity of the models is directly accounted for by the marginal likelihoods
\cite{Baele2012}. Kass \& Raftery \cite{Kass1995} developed guidelines to
interpret BF values such that $2.ln(B_{01})$ (where $B_{01} =
\frac{p(Y|M_1)}{p(Y|M_0)}$ with $Y$ is the observed data and $M_i$ the model
under consideration) must be $> 6$ to consider the evidence ``strong'' and $>
10$ to consider it ``very strong''. Here, we use PS and SS methods to
estimate and compare the marginal likelihoods of alternative assignments of
individuals to species in *BEAST. With this approach, we test whether we can
detect signal of reproductive isolation among individuals that cannot be teased
apart reliably based on morphology alone.

We investigated whether *BEAST favored models where all or some of the following
ESUs were considered distinct species: WPac, Hawaii and RedSea or actually could
not be distinguished from ESU1 (Table \ref{tab:delineationModels}). We used ESU3
and gracilis as outgroups, and to allow the estimation of the topology when all
individuals for the species tested where attributed to ESU1. In these analyses,
we selected individuals for which we had at least one mitochondrial marker and
one nuclear marker. We used data from the following markers: 16S, COI, ATP6,
c0036, c0775, ITS, LSU and H3a; and the same approach as for the phylogeny of
the entire complex, using Gblocks to remove ambiguous parts of the alignment in
ITS, PartitionFinder to find the most appropriate partition scheme and models of
molecular evolution. We unlinked clocks for all markers, used strict molecular
clocks, fixing the COI clock at 1.85\%/My \cite{Lessios2008} while estimating
the rate of the others. We ran the analysis for $50.10^{6}$ generations and
repeated it at least twice for each group.

For the estimation of the marginal likelihoods, we used the settings recommended
by Baele et al. \cite{Baele2012} with path steps of 100, and sampling along
evenly spaced quantiles of a $Beta(0.3, 1)$ distribution, but increased the
chain lengths to $3.10^{6}$. If for any runs, the absolute difference between
the marginal likelihoods estimated with the stepping stone and the path sampling
methods was higher than 1, we interpreted this difference as a failure of the
estimate to converge. We therefore discarded the results and repeated the
analysis until the difference in likelihood was lower than 1.

\begin{table}
  \centering
  \caption{Species groupings used in models of species delineation.}
  \begin{tabular}{ l l l }
    \hline
    Model                     & Number of species & ESUs attributed to ESU1 \\
    \hline 
    M1 (noWpacHawaiiRedSea)   &     2             &  WPac, Hawaii, RedSea   \\
    M2 (noWpacRedSea)         &     3             &  WPac, RedSea           \\
    M3 (noWpacHawaii)         &     3             &  WPac, Hawaii           \\
    M4 (noWpac)               &     4             &  WPac                   \\
    M5 (noHawaii)             &     4             &  Hawaii                 \\
    M6 (noRedSea)             &     4             &  RedSea                 \\
    M7 (allESU1)              &     5             &  --                     \\
    \hline
  \end{tabular}
  \label{tab:delineationModels}
\end{table}

To confirm that the differentiation in COI which delineates the putative species
using GMYC does not drive differences in marginal likelihoods between the
different models tested with *BEAST, we repeated the analyses omitting COI (for
the models M4, M5, M6 and M7).

% TODO -- Furthermore, to assess more generally the contribution of the nuclear
% data to the results, we repeated the analyses by attributing randomly the
% nuclear data.

\section{Results}

\subsection{Sequence data}

<<loci-characteristics-data, cache=TRUE, echo=FALSE, warning=FALSE>>=

@

The complete sequence data set was \Sexpr{sum(lociChar["bp_alg",
  ])}~bp (with \Sexpr{sum(lociChar["bp_alg", c("16S", "COI",
  "ATP6")])}~bp from mitochondrial genome, and
\Sexpr{sum(lociChar["bp_alg", c("c0036", "c0775", "H3a", "ITS",
  "LSU")])}~bp from the nuclear genome,
Table~\ref{tab:loci-characteristics}). Of the
\Sexpr{sum(lociChar["S_i",])} parsimony-informative sites,
\Sexpr{round(sum(lociChar["S_i", c("16S", "COI",
  "ATP6")])/sum(lociChar["S_i",])*100, 0)}\% were coming from the
mitochondrial loci (Table~\ref{tab:loci-characteristics}).

<<loci-characteristics-table, cache=TRUE, dependson='loci-characteristics-data', warning=FALSE, echo=FALSE, results='asis'>>=

@ 

<<loci-coverage-data, cache=TRUE, dependson='loci-characteristics-data', warning=FALSE, echo=FALSE>>=

@

We obtained sequence data from \Sexpr{length(unique(locWhich[,
  "Extract"]))} individuals covering the entire known geographical
range of \textit{Holothuria impatiens}. All of these individuals were
sequenced for at least one mitochondrial locus, and
\Sexpr{round(length(unique(hasMtNuc[,
  "Extract"]))/length(unique(locWhich[, "Extract"]))*100, 0)}\% were
sequenced for at least 1 nuclear locus and
\Sexpr{round(sum(table(hasMtNuc[, "Extract"]) >=
  2)/length(unique(locWhich[, "Extract"])) *100, 0) }\% for 2 or more
(Fig.~\ref{fig:loci-coverage-plot}).

<<loci-coverage-plot, cache=TRUE, dependson='loci-coverage-data', warning=FALSE, echo=FALSE, fig.keep='last', fig.width=6, fig.height=9.75, out.width='10cm', fig.cap='Representation of the loci sequenced for each individual included in the analyses.'>>=

@ 

% Here comment on results based on RAxML runs on individual loci.


\subsection{GMYC}

<<gmyc-coi-results, cache=TRUE, eval=TRUE, echo=FALSE>>=
@ 

<<gmyc-coi-vars, eval=TRUE, echo=FALSE, cache=TRUE, dependson='gmyc-coi-results'>>=
minSingle <- min(subset(gmycSumm, analysisType == "single")$mean)
maxSingle <- max(subset(gmycSumm, analysisType == "single")$mean)
minMulti <- min(subset(gmycSumm, analysisType == "multi")$mean)
maxMulti <- max(subset(gmycSumm, analysisType == "multi")$mean)
stopifnot(all(subset(gmycSumm, analysisType == "single" &
                 clock == "strict" & sequences == "allSeq")$mean == 16))
stopifnot(all(subset(gmycSumm, analysisType == "single" &
                 clock == "strict" & sequences == "noDup")$mean == 17))
allSeqStrictSingle <- gmycSumm[gmycSumm$analysisType == "single" &
                               gmycSumm$clock == "strict" &
                               gmycSumm$sequences == "allSeq", "mean"]
noDupStrictSingle <- gmycSumm[gmycSumm$analysisType == "single" &
                              gmycSumm$clock == "strict" &
                              gmycSumm$sequences == "noDup", "mean"]
@ 

The number of species estimated by the GMYC methods varied widely and ranged
from \Sexpr{minSingle} to \Sexpr{maxSingle} for the single-threshold method, and
from \Sexpr{minMulti} to \Sexpr{maxMulti} for the multi-threshold method
(Fig~\ref{fig:gmyc-coi-plot}).

<<gmyc-coi-plot, dependson='gmyc-coi-data', eval=TRUE, cache=TRUE, echo=FALSE, fig.keep='last', fig.height=4, fig.cap='Number of species estimated by single and multi-threshold GMYC methods on COI phylogenies'>>=
@ 

sGMYC produced species estimates that were more conservative and with
narrower confidence intervals (CI) than the mGMYC approach. The
maximum-likelihood estimates for the sGMYC were in most cases at the
lower limit of the CI, while the estimates from the mGMYC were at the
upper limit of the CI. For the analyses performed on all the
sequences, the CI for the estimated number of species using sGMYC and
mGMYC did not overlap.

On the genealogies estimated with a strict clock, sGMYC was less sensitive to
the type of tree prior used, as the estimated number of species was identical
for the three priors tested with \Sexpr{unique(allSeqStrictSingle)} and
\Sexpr{unique(noDupStrictSingle)} estimated species for analyses on all
haplotypes and on unique haplotypes respectively. The Yule prior produced equal
or more conservative estimates of the number of species compared to the two
types of coalescent priors.

With sGMYC, the estimated number of species was more conservative and
less sensitive to the priors used to reconstruct the genealogy when
all haplotypes were considered.

% discuss identity of species delineated.

\subsection{Species based on morphology}

Members of the impatiens group are usually characterized by having a body wall
dominated by a dark color and a lighter one. Both are considerably variable
across species and across individuals. The dark color is typically brown but
varies in lightness and can take a red or purple tint. The lighter color is
typically gray but varies in lightness from light brown to yellow or even
white. The light-colored areas, are also typically harboring a
``salt-and-pepper'' pattern with small spots of a darker coloration on the light
background.

ESU2: Easily recognizable because of (1) yellow tentacles and tube feet, (2)
round tubercles, (3) light gray/purplish background with white areas, and dark
stripes in the anterior ends, progressively becoming blotches towards posterior
end, clearly marked salt-and-pepper coloration typically more prevalent towards
posterior end. In some specimens (most common in juveniles), coloration
uniformly dark purple, or black, with only tube feet appearing yellow. Cuvierian
tubules, rarely released, long, relatively wide (about 2mm in diameter), white
with a light blue tint.

ESU3: 



\subsection{Phylogenetic relationships among the ESUs}
% TODO -- add definition of abbreviation for each ESU in figure legend if needed.
<<impatiens-tree, dependson='init-phylo', eval=TRUE, cache=TRUE, echo=FALSE, fig.keep='last', fig.height=8, fig.cap="Bayesian phylogenetic tree of the \\textit{Holothuria impatiens} complex. Vertical bars along the tips of the tree show the different ESUs found in the complex. Axis on the bottom in My.">>=
@

The topologies obtained with BEAST and RAxML analyses were very similar
(Fig.~\ref{fig:impatiens-tree} and Supplementary Material~\ref{}), and the
clades identified were found for all the loci analyzed (Supplementary
Material~\ref{}).

The topologies were highly supported at least up to the node corresponding to
the most recent common ancestors of the species delineated with the most
conservative result of the GMYC method. 

The \textit{impatiens} complex comprises two clades that diverged
approximately 10 MYA (Fig.~\ref{fig:impatiens-tree}). Clade I has the widest
geographic distribution and is found in the Mediterranean Sea, the Caribbean
basin, the Eastern-Pacific and across the Indo-West Pacific, including the Red
Sea. Clade II is restricted to the Indo-West Pacific (including the Red Sea).

The phylogeny also highlights that several speciation events in the complex were
concurrent throughout the diversification history of the complex (e.g., the first
speciation event in each clade about 9.5 MYA).

The youngest pair of sister species delineated by GMYC are estimated to have
diverged 1.8 MYA. The three delineated species that diverged the most recently
(ESU1, WPac and Hawaii) have overlapping distributions
(Figure~\ref{fig:impatiens-map-group2}).

%% Look into Hernandez-Molina2014 for timing of fluctuation/closures to compare
%% with phylogeny.

<<impatiens-map-WA, dependson='init-map', eval=TRUE, cache=TRUE, echo=FALSE, fig.keep="last", message=FALSE, warning=FALSE, results="hide", fig.cap="Distribution maps of the ESUs in Eastern Pacific and Caribbean">>=
@ 

<<impatiens-map-group1, dependson='init-map', eval=TRUE, cache=TRUE, echo=FALSE, fig.keep="last", message=FALSE, warning=FALSE, results="hide", fig.cap="Distribution map for the ESUs in group 1.">>=
@ 

<<impatiens-map-group2, dependson='init-map', eval=TRUE, cache=TRUE, echo=FALSE, fig.keep="last", message=FALSE, warning=FALSE, results="hide", fig.cap="Distribution map for the ESUs in group 2.">>=
@ 

\subsection{Species limits}

Both PS and SS gave almost identical estimates of the marginal likelihoods, and
here, we only report the results from the stepping-stone approach (see
supplementary material \ref{sm:pathsampling} for the PS results). The model with
all the putative species identified by GMYC has the highest marginal likelihood,
and this holds for both independent fittings
\ref{fig:starbeast-summary}. However, the difference in marginal likelihoods for
the model that considers the Wpac and Hawaii putative species as not being
reproductively isolated from ESU1 is only below the threshold needed to consider
the evidence strong.

<<starbeast-summary, dependson='init-phylo', eval=TRUE, cache=TRUE, echo=FALSE, fig.keep='last', message=FALSE, warning=FALSE, results='hide', fig.scap='Differences in log-marginal likelihoods for various models of species assignment to individuals belonging to the ESU1 complex.', fig.cap='Differences in log-marginal likelihoods (estimated by stepping-stone sampling) for various models of species assignment to individuals belonging to the ESU1 complex. The results are reported for two independent fittings of the model (dark and light gray bars). The differences are calculated compared to the second best model which provides the minimum (conservative) difference between the best fit model and the other models. The dashed line represent a difference in log-marginal likelihood of 5 that provides a Bayes Factor value of 10, considered as \"strong evidence\" according to  Kass \\& Raftery \\cite{Kass1995}. The best-fit model includes all the putative species identified by the GMYC approach.'>>=
@ 


% cryptic species because indistinguishable, indistinguishable because similar
% looking or too much variation to make sense of it

% information about their biology gathered on the supposition it's a single
% species throughout its range

% from Volger+2012, most accurate COI divergence rate for echinoderms,
% 3.7(+/-)0.8%/Myr, from Lessios paper about Panamanian isthmus.

% look into splitstree (already installed into ~/Software folder), this allows
% to test for the robustness of signal in minimum spanning tree (see Volger+2012
% for details)

% look into \phi_{ST}, Fu's F, Tajima's D (reading the Arlequin manual)

% one of the reason, H impatiens has not been recognized as a species complex
% earlier, is due to the fact that some ESUs are widespread and others are
% overlapping. One of the reasons, these ESUs can overlap is that there was an
% ancestral widespread form which gave rise to the other species by parapatric
% speciation (e.g., as in Machel's hermit crabs), the alternative view is that
% each species recently extended their ranges. Should I try to use Lagrange? or
% Beast to reconstruct the phylogeographic history of the species to tease apart
% between these 2 hypotheses?
% to identify biogeographic regions, see if there are consistent phylogeographic
% patterns among ESUs

\subsection{Diversity}
% how many clades
% how many specimens per clades


\section{Discussion}

\subsection{Species delineation using GMYC}

Despite being widely used to delineate species, the sensitivity of the GMYC
method to the factors affecting the shape of the tree used have not been
investigated thoroughly. Tree shape can be affected by biological factors or by
the type of analysis used to build the tree. Simulated data have been used to
test the influence of diversification rates, effective population sizes, and
migration rates on GMYC accuracy and precision \cite{Papadopoulou2008,
  Esseltyn2012}. Some of the effects of the methodological choices have been
evaluated by the authors of the method \cite{Pons2006, Monaghan2009,
  Fujisawa2013} and by others \cite{Talavera2013}, but the conclusions might be
difficult to generalize depending on the nature of the dataset. Here, we showed
that keeping all COI sequences, using a relaxed molecular clock and a Yule tree
prior, led to the most realistic number of delineated species.

This combination of priors led to overestimating the number of species when it
was tested by the authors of the method \cite{Monaghan2009}. On the other hand,
Talavera et al \cite{Talavera2013} did not find any notable differences using
trees estimated with different sets of priors. Our dataset differs from these
studies by including a younger radiation (here: less than 10 Ma,
\cite{Monaghan2009}: 4 insect orders; \cite{Talavera2013} 5 Lepidoptera
families), less species (here: 14; \cite{Monaghan2009} about 370;
\cite{Talavera2013} 176), and more individuals per species (here: about 14
ind/species; \cite{Monaghan2009}: about 3 ind/species; \cite{Talavera2013} about
7 ind/species).

% Esseltyn+2012 also used both all sequences & only unique, but didn't provide
% the differences in results

Talavera et al \cite{Talavera2013} did not find that using all haplotypes or
only the unique haplotypes affected the number of species estimated with the
GMYC method. Here, we found that removing the duplicated sequences overestimated
the number of species and decreased the precision of the estimates. Keeping
identical sequences in the analysis, overestimates the effective population size
parameters that BEAST uses. In turns, these overestimated sizes spuriously
increase the time to coalescence for the lineages, making the estimation of the
inflection point between species- and population-level coalescent events earlier
in the tree and more difficult, leading to overestimation in the number of
putative species and the decrease in accuracy.

GMYC was developed with the intent of automating the estimation of the number of
species from large-scale single-locus sampling efforts. In this context, a
reasonable amount of inaccuracy or imprecision from the true number of species
might be acceptable. The advantages of the method, namely providing a rapid and
objective assessment of the number of species, outweigh the possible deviations
from the true value. However, in the context of a taxonomic study where the goal
is to identify clearly species limits among a few hundred individuals, 
should be used in the method

GMYC is suitable for obtaining a rough estimate of the number of species among a
sample, but if used in systematics study to delineate species, care needs to be
taken to use the appropriate priors for the particular type of
dataset. Additionally, for this type of studies, additional lines of evidence
need to be used to validate the putative species delineated by GMYC (this study,
\cite{Marshall2011a, Esseltyn2012}).

% talk about time spent on phylogeny between coalescent and speciation
% processes.

% Effect of time/age of the phylogeny

% Method in Pons+2006 and 
% Examples: Volger+2008 (COTS paper), Marshall+2011 (cicadas in New Zealand);
% Henrich+2010 (water beetles in Australia)
% See: Esselstyn+2012 as it includes simulation study of statistical power of
% this approach; Lohse2009 for a critic

\subsection{Using *BEAST and model selection for species delineation}

%% From the BEAST book
% It can happen that the species assignment is incorrect, which results in co-
% alescent events higher up in the species tree than if the species assignment is
% correct. As a result population size estimates will be unusually high. So,
% relative high population size estimates may indicate incorrect assignments, but
% it can also indicate the existence of a cryptic species

% Low PP on some nodes, probably from using low number of loci (and duplicated
% sequences). This low number of loci, is also not enough to obtain accurate
% population estimates (see Harris+2013,10.1111/evo.12280).

\subsection{Coloration, taxonomy, species identity}

%% Coloration & taxonomy
% This paper http://dx.doi.org/10.1016/j.ympev.2014.01.017
% is an example of potential overspliting based on coloration or really recent
% species not yet distinguishable genetically and/or morphologically.

%% Taxonomic consequences
% impatiens should be elevated to sub-genus level
% no other Thymiosycia (except from gracilis) are in this group.

%% Differences in ecology among the species
% ESU1 not picky but lagoons
% ESU2 
% Wpac, gracilis and ESU3 continental (hence distribution)

%% Diversification of the complex is really ~20 MY really fits with idea that
%% most of the reef-associated fauna diversified with corals see citations
%% within Williams et al 2011.

%% Compared to other species (e.g., Cephalopholis argus see Gaither et al 2011;
%% many other cukes), there is no Indian Ocean -- Pacific Ocean divide here.
%% How do range distributions in impatiens can shed light on the debate on the
%% origins behind the patterns of species distribution in the IWP?

%% Talk about how IO morphs are different from Pacific even though there is no
%% genetic differentiation.

% Add something about how differences in colors among ESUs might be a result of
% the same type of proteins than GRP.

% Many cryptic species = change in patterns of biodiversity
% - find examples of other species that show similar patterns indicative of small
%   scale endemism
% - potential to redefine hotspots
% - 

The ability for taxonomists to be able to recognize populations of
reproductively isolated individuals as species requires differentiation at three
levels: morphological, genetically

\subsection{Patterns of genetic diversity and geographical distribution}

% Bowen et al 2013 -- TREE -- multiple examples of species with no
% differentiation across large geographic scales

%% Geography of speciation
% Do closely related species are close geographically or are they isolated? Is
% there evidence for range contraction (sister species between distant parts of
% the IWP)?

% one of the difference between marine and terrestrial speciation is that the
% dissociation between geogrpahical isolation and absence of gene flow is
% probably stronger in the sea.

%% New genomic methods will allow better characterization of demography of the
%% species, and finer understanding of what happened in this complex.

%% Possible scenarios:
% - selection under uniform selection: different mutations are fixed under same
% environment in populations evolving allopatry, alternate mutations cause
% postzygotic isolation by DM incompatibilities (Sobel et al 2010)

%% "Geographic ranges are the product of both ecological and historical factors"
%% "Historical processes leading to allopatry will often give way to ecological
%% processes as populations adapt to different habitats."
%% (Sobel et al 2010)

% Use of ecological niche models to see if there are differences in habitat
% preference.

% According to \cite{Renema2008}, the modern IWP fauna has its origin in the
% Miocene. Generation of hotspots associated with tectonic events, so importance
% of abiotic factors.

% Therefore, as it appears that the majority of speciation events are
% allopatric (\cite{Coyne2004}), recent changes in distribution and population
% sizes must be prevalent in the \textit{H. impatiens} complex to explain current
% distribution patterns. These changes have implications regarding the rapid
% evolution of reprodutive isolation barriers and ecological adaptations.

% tiger and tigerRedSea example of disjoint species distribution see
% Williams+2011. 

% relationship between reproductive isolation and genetic
% differentiation for sympatry and allopatric species (review
% Rabosky&Matute PNAS).

A striking pattern among the members of the \textit{H. impatiens} complex, is
the differences in the geographical ranges among its members. Some species in
the complex are restricted in their range to an archipelago (e.g., Hawaii or the
Galapagos) while others can be found across the Indian and most of the Pacific
Oceans with almost no genetic differentiation (e.g., ESU1, ESU2). The origins of
these differences can stem from differences in demographics history. For
instance, the low genetic diversity combined with the large range of ESU1
suggest a recent rapid expansion of its range. This is corroborated by the XXX
values from the *BEAST analyses.

If for some species in the complex the range restriction is caused by
geographical and oceanographic conditions limiting the export of propagules
(e.g., Red Sea, Mediterranean), others have ranges that overlap with (e.g.,
Wpac, ESU3, gracilis) or are included within (e.g. Hawaii) the range of a
species with a wide distribution (ESU1). In these cases, dispersal limitation
seems unlikely to restrict their ranges, and instead, adaptation to ecological
conditions are more likely to limit their distribution. It is interesting to
note that gracilis, Wpac, and ESU3 are all associated with continental
areas. For instance, ESU3 is found in Madagascar and Tanzania but was not found
in the oligotrophic areas of Mayotte or any of the Scattered islands sampled. It
is however found in Palau and the Ryukyus but the former is known to be
characterized with extensive eutrophic lagoons, and the latter is under the
influence of upwellings. % *TODO* check these claims.

restricted ranges with a distribution that overlaps with species having a broad
range (e.g., Hawaii overlapping with ESU1). This suggests that

However, differences in the ecology of these species must also play an important
role. The ecological conditions found in the Hawaiian Gala

One of the main questions when studying patterns of speciation is whether modern
species distributions reflect their distributions at the time of speciation. For
marine organisms, and in particular within the Indo-Pacific, this assumption is
particularly difficult to hold as (1) current barriers for dispersal are few in
numbers; (2) the dispersal ability of most marine invertebrates is high; (3) the
current landscape of the Indo-Pacific is relatively recent. For instance, the
faunal breaks observed among species composition of reef-building corals are
only poorly correlated with present-day conditions but are much more strongly
associated with geological features \cite{Keith2013}.

Bowen et al \cite{Bowen2013} suggest that sympatric speciation is possible among
marine organisms. In particular, they cite Rocha \cite{Rocha2008} that
investigated patterns of speciation among grunts (\textit{Haemulon}) in the
Caribbean and the Eastern Pacific that show closely related species occurring in
sympatry. Authors in this study invoke the possible role of the sounds members
of this genus produce in assortative mating.


\subsection{Rate of diversification, reproductive isolation}

%% Short review on how the rates of secondary sympatry is much faster in
%% impatiens compared to other examples.
% - in Quenouille+2011, fish take at least 4 MYa to reach secondary sympatry
% - Look into \cite{Lessios1999} to see how the timing in Eucidaris compares to
% the one H. impatiens

%% Knowlton 2000 says that for alpheids, all species are older than
%% closure of isthmus of Panama (except for some fish, and snails
%% [examples given are however, from continental setting])
%% - see Puritz to see what they say exactly about timing of speciation.
%% - \cite{Williams2011e} in Lunella, "sympatric species occur only along coastlines,
%%   while species occuring in island arrays occur in allopatry".

%% Traditionally 2 views:
%% - small populations -- speed up divergence?
%% - local adaptation -- role of ecology and interplay with selection on GRPs

Two kinds of models have been proposed to explain divergence in marine 
invertebrates in absence of clear geographical barriers: founder effect \cite{},
and selection in response to local adaptation \cite{Bowen2013}. These two forces
being non-mutually exclusive. Here, we argue that in either case, evolutionary
and demographic forces interplay with selection on gamete recognition proteins
to favor divergence.

In the case of founder effects (parapatric speciation), where a small population
establishes itself in the periphery (and therefore being connected with limited
gene flow) from the ancestral population, the genetic diversity of these
individuals will be limited. The success of the establishment of this peripheral
population will depend on its density and its genetic make up that will allow
for its local adaptation to ecological conditions in the new environment, and
will ensure the compatibility of the genotypes for the gamete recognition
proteins. In sea urchins, Levitan \& Ferrel have shown that at low density,
males and females with identical genotypes for bindin, have higher reproductive
success \cite{Levitan2006}. Therefore, if the pool of individuals that colonized
the peripheral location harbors genotypes that are non-representative of the
ancestral populations, and/or if the genotypes are in linkage-disequilibrium
with genes being selected for in the new environment, GRPs will drive divergence
between populations, leading to reproductive isolation.




%% in any case, probably small effective population sizes due differences in
%% reproductive success, 


%% From van Doorn et al 2009 "Once a mating preference for locally adapted
%% partners has evolved, sexual selection reinforces assortative mating and
%% lowers the fitness of hybrids." How does density dependence in fecundation
%% success interplays with/changes this statement?
%% One can imagine that if particular genotype associated with high fitness in
%% particular habitat, abundance is higher, 

%% Review papers on GRP to see if they are considered a Fisherian sexual
%% selection trait







% Speed of speciation/secondary sympatry
% - how they managed to accomplish that
% - signature of population expansion?
% - implication for identification of geographical boundaries in distribution
% - role of pre- and post-zygotic barriers
% - role of selection?

\section{Conclusions}

% museum collections need to be future proof


\section{Acknowledgments}

CIPRES portal.
UF HPC, HiPerGator.
wesanderson
knitr
polymode + ESS

\bibliographystyle{plos2009}

\bibliography{impatiens_phylogeography}

\section{Supplementary materials}

\begin{table}
  \centering
  \caption{The subscript number corresponds to the to the codon position in the
    alignment. The "real" codon position is offset by 2 for COI, H3a and ATP6
    (i.e., $COI_1$ is 3rd codon position)}
  \begin{tabular} { l l l }
    1      & $GTR+I+G$    & $16S, ATP6_2$ \\
    2      & $HKY+I+G$    & $ATP6_1, COI_1$\\ 
    3      & $HKY+I+G$    & $ATP6_3, c0775$\\ 
    4      & $GTR+I+G$    & $c0036$\\ 
    5      & $GTR+I+G$    & $COI_2, H3a_2$ \\
    6      & $HKY+I$      & $COI_3$\\ 
    7      & $HKY$        & $H3a_1, LSU$ \\ 
    8      & $JC$         & $H3a_3$ \\
    8      & $HKY+G$      & ITS \\       
  \end{tabular}
\end{table}


<<sm-starbeast-summary, dependson='starbeast-summary', eval=TRUE, cache=TRUE, echo=FALSE, fig.keep='last', message=FALSE, warning=FALSE, results='hide', fig.scap='Differences in log-marginal likelihoods (estimated by path sampling) for various models of species assignment to individuals belonging to the ESU1 complex.', fig.cap='Differences in log-marginal likelihoods (estimated by path sampling) for various models of species assignment to individuals belonging to the ESU1 complex. The results are reported for two independent fittings of the model (dark and light gray bars). The differences are calculated compared to the second best model which provides the minimum (conservative) difference between the best fit model and the other models. The dashed line represent a difference in log-marginal likelihood of 5 that provides a Bayes Factor value of 10, considered as \"strong evidence\" according to  Kass \\& Raftery \\cite{Kass1995}. The best-fit model includes all the putative species identified by the GMYC approach.'>>=
@ 


%% Notes about the biology of the species
%% - EP: discharge Cuv tubules \cite{Bakus1974}, ``relatively very active''
%% \cite{Bakus1974}, 
%% - Egypt: tables appear earlier than buttons in juveniles, look for pl.8 fig4-5,
%% and pl.9 2 in Savigny \cite{Mortensen1902}10.1111/j.1096-3642.1926.tb00326.x

%% READ \cite{Cutress1996}, seems like there is a lot of information about
%% ontogenetic changes in the shape of the ossicles.
%% ``Buttons in OM dermis of the 195 mm specimen from Ceylon, however, are
%% distinctly smaller.'' 

\end{document}




